src/core/simulation_results.jl

```diff
@@ -274,7 +274,10 @@ function load_simulation_results(
     variable::Array;
     kwargs...,
 )
-    results_folder = sim_output.results_folder
+    results_folder = joinpath(sim_output.results_folder, stage_name)
+    if !isdir(results_folder)
```

Comment:
If you use `mkpath` then you can avoid checking for existence.  Just an FYI if prefer conciseness.